agent/src/workload.rs

```diff
@@ -81,19 +84,62 @@ impl Workload {
             control_interface,
         }
     }
+    // [impl->swdd~agent-create-control-interface-pipes-per-workload~1]
+    fn set_control_interface(
+        &mut self,
+        pipes_channel_context_info: Option<PipesChannelContextInfo>,
+    ) {
+        if let Some(control_interface) = self.control_interface.take() {
+            control_interface.abort_pipes_channel_task()
+        }
+        self.control_interface = match pipes_channel_context_info {
+            Some(info) => info.make_context(),
+            None => None,
+        };
+    }
+
+    fn update_control_interface(
+        &mut self,
+        spec: &Option<WorkloadSpec>,
+        pipes_channel_context_info: Option<PipesChannelContextInfo>,
+    ) {
+        match (&mut self.control_interface, spec) {
+            (None, None) => (),
+            (Some(_current), None) => self.set_control_interface(None),
+            (None, Some(_new)) => self.set_control_interface(pipes_channel_context_info),
+            (Some(current), Some(new_spec)) => {
+                match (
+                    current.get_api_location().to_str(),
+                    pipes_channel_context_info.as_ref(),
+                ) {
+                    (None, None) => (),
+                    (None, Some(_new)) => self.set_control_interface(pipes_channel_context_info),
+                    (Some(_current_api_location), None) => self.set_control_interface(None),
+                    (Some(current_api_location), Some(new_pipes_channel_context_info)) => {
+                        let new_control_interface_location = new_spec
+                            .instance_name
+                            .pipes_folder_name(new_pipes_channel_context_info.get_run_folder());
+
+                        if !current_api_location
+                            .contains(new_control_interface_location.to_str().unwrap())
+                        {
+                            self.set_control_interface(pipes_channel_context_info);
+                        }
+                    }
+                };
+            }
+        }
```

Comment:
I'm not sure if the `match` here is making things easy to understand. Maybe we can rewrite as follows:
```suggestion
        if self.control_interface.is_none() {
            self.set_control_interface(pipes_channel_context_info);
        } else if pipes_channel_context_info.is_none {
            self.set_control_interface(None);
        } else if is_control_interface_changed(self.control_interface, pipes_channel_context_info) {
            self.set_control_interface(pipes_channel_context_info);
        }
```

We additionally need the `is_control_interface_changed` function.

Do we also need to unit test the two functions? The logic seems relatively complex, maybe we should pinpoint it with some tests to avoid regressions.